internal/actions/sync_branch.go

```diff
@@ -420,6 +423,36 @@ func syncBranchRebase(
 	return nil, nil
 }
 
+func fetchUpstreamCommits(repo *git.Repo, tx meta.WriteTx, branch meta.Branch) (string, error) {
+	parent := branch
+	for parent.Parent.Name != "" {
+		parent, _ = tx.Branch(parent.Parent.Name)
+	}
+
+	if _, err := repo.Git("fetch", "origin", parent.Name); err != nil {
+		return "", errors.WrapIff(err, "failed to fetch %q from origin", parent.Name)
+	}
+	commitHash, err := repo.RevParse(&git.RevParse{Rev: "FETCH_HEAD"})
+	if err != nil {
+		return "", errors.WrapIff(err, "failed to read the commit hash of %q", parent.Name)
+	}
+	return commitHash, nil
+}
+
+// findMergeCommitWithGitLog looks for the merge commit for a specified PR.
+//
+// Usually, GitHub should set which commit closes a pull request. This is known to be not that
+// reliable. When we cannot find a merge commit for a closed PR, we try to find if any commit in the
+// upstream closes the pull request as a fallback.
+func findMergeCommitWithGitLog(repo *git.Repo, upstreamCommit string, branch meta.Branch) (string, error) {
+	cis, err := repo.Log(git.LogOpts{RevisionRange: branch.Name + ".." + upstreamCommit})
```

Comment:
Can we use the more explicit syntax here rather than the double dot? I don't always remember what the double dot means. Also would be helpful either way to have a comment explaining what we're querying.
```suggestion
    // Find all commits that have been merged into the trunk since this branch
	cis, err := repo.Log(git.LogOpts{RevisionRange: []string{upstreamCommit, "^" + branch.Name}})
```

(^assuming you incorporate my suggested change below about `RevisionRange` being a slice)

Reply:
Done.